File: lib/src/main/java/com/dhy/retrofitrxutil/IDisposable.java
package com.dhy.retrofitrxutil;

import android.content.Context;
import android.support.annotation.NonNull;

import io.reactivex.disposables.Disposable;

public interface IDisposable {
    void registerDisposable(@NonNull Context context, @NonNull Disposable disposable);

    void onComplete(Context context, @NonNull Disposable disposable);
}

File: lib/src/main/java/com/dhy/retrofitrxutil/IDisposable.kt
package com.dhy.retrofitrxutil

import android.content.Context
import io.reactivex.disposables.Disposable

/**
 * must call  DisposableHandler.onDestroy(this) in Activity.onDestroy
 * */
interface IDisposable {
    fun registerDisposable(context: Context, disposable: Disposable) {
        DisposableHandler.registerDisposable(context, disposable)
    }

    fun onComplete(context: Context, disposable: Disposable) {
        DisposableHandler.onComplete(context, disposable)
    }
}